### PR TITLE
[CWS] use one model per platform

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

This PR is a first part of PRs aiming to introduce a windows support for CWS. This PR does the following : 

- Modify the generator so that it can handle multiple model files.
- Extract common structures that can be used for windows and linux models
- Introduce a windows model in a dedicated model_windows.go file
- Rework the event call path so that it can handle either linux or windows events

What is remaining : 

- Generation of the windows documentation
- Windows event serialization
- Common API server for both linux and windows, gRPC sockets

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: docs/cloud-workload-security/secl.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "event_types": [
     {
       "name": "*",
-      "definition": "",
+      "definition": "SECLDoc[event.async] Definition:`True if the syscall was asynchronous`",
```
Comment (paulcacheux) [inline docs/cloud-workload-security/secl.json:5]: I think this indicates an issue if this changes no ?

File: pkg/security/secl/model/accessors_linux.go [hunk 2]
```diff
@@ -4,8 +4,8 @@
 // Copyright 2022-present Datadog, Inc.
 // Code generated - DO NOT EDIT.
 
-//go:build unix
```
Comment (paulcacheux) [inline pkg/security/secl/model/accessors_linux.go:7]: the fact that this was unix allowed this to build successfully on macOS as well, i.e. making work the dd-go rule checker local tests for dev there. Why do we need to change this ?

File: docs/cloud-workload-security/secl.json (hunk 1 above)
Comment (safchain) [reply docs/cloud-workload-security/secl.json:5]: Yes, I'm going to have a look

**Review by drichards-87 — APPROVED** [review]
Approved from Docs.

File: pkg/security/secl/model/accessors_linux.go (hunk 2 above)
Comment (safchain) [reply pkg/security/secl/model/accessors_linux.go]: thanks Paul, reverted, back to unix tag

File: docs/cloud-workload-security/secl.json (hunk 1 above)
Comment (safchain) [reply docs/cloud-workload-security/secl.json]: It should be ok now

Comment (paulcacheux) [reply docs/cloud-workload-security/secl.json]: thanks !

File: pkg/security/secl/model/accessors_linux.go (hunk 2 above)
Comment (paulcacheux) [reply pkg/security/secl/model/accessors_linux.go]: thanks !

**Review by paulcacheux — APPROVED** [review]

File: pkg/process/events/consumer/events_consumer_windows.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+		EventType:      model.NewEventType(event.GetEventType().String()),
+		CollectionTime: event.Timestamp,
+		Pid:            entry.Pid,
+		ContainerID:    entry.ContainerID,
+		Ppid:           entry.PPid,
+		Exe:            entry.FileEvent.PathnameStr, // FileEvent is not a pointer, so it can be directly accessed
+		Cmdline:        cmdline,
+		ExecTime:       entry.ExecTime,
+		ExitTime:       entry.ExitTime,
+		ExitCode:       event.Exit.Code,
```
Comment (mbotarro) [inline pkg/process/events/consumer/events_consumer_windows.go:36]: Just for my understanding, is there any correspondent information to `Username`, `Group`, `ForkTime`, `UID` and `GID` that we could use on Windows systems?

**Review by mbotarro — APPROVED** [review]
It looks great for changes in `pkg/proces`!

File: pkg/process/events/consumer/events_consumer_windows.go (hunk 3 above)
Comment (paulcacheux) [reply pkg/process/events/consumer/events_consumer_windows.go:36]: yes ! the goal of this PR is really to split the 2 models, the next step is to add what makes sense for windows, including a different user/group model, different file model etc

Comment (mbotarro) [reply pkg/process/events/consumer/events_consumer_windows.go:36]: that makes sense! thanks for clarifying it, @paulcacheux !!

**Pull request merged**